lms/djangoapps/bulk_email/tasks.py

```diff
@@ -387,25 +393,40 @@ def _filter_optouts_from_recipients(to_list, course_id):
     return to_list, num_optout
 
 
-def _get_source_address(course_id, course_title):
+def _get_source_address(user_pk, course_id, course_title):
     """
     Calculates an email address to be used as the 'from-address' for sent emails.
 
-    Makes a unique from name and address for each course, e.g.
+    Makes a unique from name and address for each course,
 
         "COURSE_TITLE" Course Staff <coursenum-no-reply@courseupdates.edx.org>
 
+    The string is localized based on the user's language preference.
     """
     course_title_no_quotes = re.sub(r'"', '', course_title)
 
     # For the email address, get the course.  Then make sure that it can be used
     # in an email address, by substituting a '_' anywhere a non-(ascii, period, or dash)
     # character appears.
-    from_addr = u'"{0}" Course Staff <{1}-{2}>'.format(
-        course_title_no_quotes,
-        re.sub(r"[^\w.-]", '_', course_id.course),
-        settings.BULK_EMAIL_DEFAULT_FROM_EMAIL
+
+    try:
+        user = User.objects.get(pk=user_pk)
```

Comment:
I also worry about doing multiple queries on each email. I'd rather get user preferences upfront.

A few months ago, we had a performance issue where we were running a bunch of queries per email. In order to informally test the performance of my fix, I ran [this test](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/bulk_email/tests/test_email.py#L182-L186) , but with 10,000 students instead of 10, and timed how long it took to do the fake send. That way you can get a sense of the performance impact of your changes.